source/elements/oneDAL/source/programming_model/computational_modes.rst

```diff
@@ -5,11 +5,18 @@ Computational modes
 -----
 Batch
 -----
-
+In the batch processing mode, the algorithm works with the entire data set to produce the final
+ result. A more complex scenario occurs when the entire data set is not available at the moment
+ or the data set does not fit into the device memory.
 ------
 Online
 ------
-
+In the online processing mode, the algorithm processes a data set in blocks streamed into device
+ memory by doing incrementally updating partial results, which are finalized upon processing of
+ the last data block.
 -----------
 Distributed
 -----------
+In the distributed processing mode, the algorithm operates on a data set distributed across
```

Comment:
we may need to extend the def once we decide on the distributed mode in the lib

Reply:
I've catch this comment for further discussions